### PR TITLE
Convert to using pyproject.toml as much as possible.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Convert to using pyproject.toml as much as possible. This is both to follow current best practices and also to try to fix recent setuptools_scm issues.

Also fixed two things in the actions yaml that seem to have fixed the MacOS builds:
- fetching the full-depth repo (without this calls to `git describe` fail)
- calling `git describe` and `git version` right before the install step. This seems to fix a problem on Mac builds where setuptools_scm's call to git was timing out. I don't really understand why...

### Build/CI Change Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
+test = ["coverage", "pre-commit", "pytest", "pytest-cov"]
+doc = ["matplotlib", "pypandoc", "sphinx"]
+dev = ["pyradiosky[all,test,doc]"]
+
+[project.urls]
+Repository = "https://github.com/RadioAstronomySoftwareGroup/pyradiosky"
+Documentation = "https://pyradiosky.readthedocs.io/"
+
+# Note that this approach is "discouraged" in favor of [project.scripts], but
+# that requires entry-point functions, which would require some refactoring.
```
Comment (steven-murray) [inline pyproject.toml:56]: I do highly recommend it though!

**Review by steven-murray — APPROVED** [review]
Approving, though noting that using entry points is very nice, and also that `ruff` is nicer and faster than flake8 these days (and configured in pyproject.toml as well)

File: pyproject.toml (hunk 1 above)
Comment (mkolopanis) [reply pyproject.toml:56]: I second this! but otherwise also feel like this PR is good. And am a ruff advocate as well.

**Comment by mkolopanis** [comment]
we may want to set `SETUPTOOLS_SCM_DEBUG=1` when performing the install for a macos build to see if there is any interesting output

File: pyproject.toml (hunk 1 above)
Comment (bhazelton) [reply pyproject.toml:56]: I think we should do this, I was just going to do it in a different PR.

Comment (bhazelton) [reply pyproject.toml:56]: Same with ruff.

Comment (bhazelton) [reply pyproject.toml:56]: I made issues for both of these so we won't lose track of them.

**Review by mkolopanis — APPROVED** [review]

**Pull request merged**